src/Compilers/CSharp/Portable/BoundTree/BoundDecisionDag.cs

```diff
@@ -206,101 +206,16 @@ BoundDecisionDagNode makeReplacement(BoundDecisionDagNode dag, Func<BoundDecisio
-            for (int i = 0; i < allStates.Length; i++)
-            {
-                stateIdentifierMap.Add(allStates[i], i);
-            }
-
-            int nextTempNumber = 0;
-            var tempIdentifierMap = PooledDictionary<BoundDagEvaluation, int>.GetInstance();
-            int tempIdentifier(BoundDagEvaluation e)
-            {
-                return (e == null) ? 0 : tempIdentifierMap.TryGetValue(e, out int value) ? value : tempIdentifierMap[e] = ++nextTempNumber;
-            }
-
-            string tempName(BoundDagTemp t)
-            {
-                return $"t{tempIdentifier(t.Source)}{(t.Index != 0 ? $".{t.Index.ToString()}" : "")}";
-            }
 
             var resultBuilder = PooledStringBuilder.GetInstance();
             var result = resultBuilder.Builder;
 
             foreach (var state in allStates)
             {
-                result.AppendLine($"State " + stateIdentifierMap[state]);
-                switch (state)
-                {
-                    case BoundTestDecisionDagNode node:
-                        result.AppendLine($"  Test: {dumpDagTest(node.Test)}");
-                        if (node.WhenTrue != null)
-                        {
-                            result.AppendLine($"  WhenTrue: {stateIdentifierMap[node.WhenTrue]}");
-                        }
-
-                        if (node.WhenFalse != null)
-                        {
-                            result.AppendLine($"  WhenFalse: {stateIdentifierMap[node.WhenFalse]}");
-                        }
-                        break;
-                    case BoundEvaluationDecisionDagNode node:
-                        result.AppendLine($"  Test: {dumpDagTest(node.Evaluation)}");
-                        if (node.Next != null)
-                        {
-                            result.AppendLine($"  Next: {stateIdentifierMap[node.Next]}");
-                        }
-                        break;
-                    case BoundWhenDecisionDagNode node:
-                        result.AppendLine($"  WhenClause: " + node.WhenExpression?.Syntax);
-                        if (node.WhenTrue != null)
-                        {
-                            result.AppendLine($"  WhenTrue: {stateIdentifierMap[node.WhenTrue]}");
-                        }
-
-                        if (node.WhenFalse != null)
-                        {
-                            result.AppendLine($"  WhenFalse: {stateIdentifierMap[node.WhenFalse]}");
-                        }
-                        break;
-                    case BoundLeafDecisionDagNode node:
-                        result.AppendLine($"  Case: {node.Label.Name}" + node.Syntax);
-                        break;
-                    default:
-                        throw ExceptionUtilities.UnexpectedValue(state);
-                }
+                result.Append(state.GetDebuggerDisplay());
```

Comment:
Should this be AppendLine or does GetDebuggerDisplay already include a newline?

Reply:
Have moved the AppendLine to this method as suggested in https://github.com/dotnet/roslyn/pull/53892#discussion_r645909816